From b3b01081a7c6fda663d3fc4d8dbc4665df37dccc Mon Sep 17 00:00:00 2001
From: Pierre Labastie <pierre.labastie@neuf.fr>
Date: Sun, 10 May 2020 15:48:59 +0200
Subject: [PATCH 1/2] FROMLIST LFS: Fix several build failures

- some introduced by glibc-2.26
- some introduced by gcc-10
- also fix an ignored error when gpm.h is not already in
  /usr/include
---
 src/daemon/open_console.c  |  4 ++++
 src/headers/daemon.h       |  2 +-
 src/prog/display-buttons.c |  2 +-
 src/prog/display-coords.c  |  2 +-
 src/prog/get-versions.c    |  2 +-
 src/prog/gpm-root.y        | 12 ++++--------
 6 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/src/daemon/open_console.c b/src/daemon/open_console.c
index 98297c9..0948501 100644
--- a/src/daemon/open_console.c
+++ b/src/daemon/open_console.c
@@ -23,6 +23,10 @@
 #include <sys/stat.h>               /* stat()            */
 #include <sys/ioctl.h>              /* ioctl             */
 
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>          /* major() w/newer glibc */
+#endif
+
 /* Linux specific (to be outsourced in gpm2 */
 #include <linux/serial.h>           /* for serial console check */
 #include <asm/ioctls.h>            /* for serial console check */
diff --git a/src/headers/daemon.h b/src/headers/daemon.h
index a8936ad..949eb84 100644
--- a/src/headers/daemon.h
+++ b/src/headers/daemon.h
@@ -180,7 +180,7 @@ extern struct mouse_features  mouse_table[3],
 extern Gpm_Type         mice[];
 extern Gpm_Type         *repeated_type;
 
-time_t                  last_selection_time;
+extern time_t                  last_selection_time;
 
 
 
diff --git a/src/prog/display-buttons.c b/src/prog/display-buttons.c
index de8e5b2..155c6c2 100644
--- a/src/prog/display-buttons.c
+++ b/src/prog/display-buttons.c
@@ -36,7 +36,7 @@
 #include <stdio.h>            /* printf()             */
 #include <time.h>             /* time()               */
 #include <errno.h>            /* errno                */
-#include <gpm.h>              /* gpm information      */
+#include "headers/gpm.h"              /* gpm information      */
 
 /* display resulting data */
 int display_data(Gpm_Event *event, void *data)
diff --git a/src/prog/display-coords.c b/src/prog/display-coords.c
index ed15c8a..8d552ba 100644
--- a/src/prog/display-coords.c
+++ b/src/prog/display-coords.c
@@ -37,7 +37,7 @@
 #include <stdio.h>            /* printf()             */
 #include <time.h>             /* time()               */
 #include <errno.h>            /* errno                */
-#include <gpm.h>              /* gpm information      */
+#include "headers/gpm.h"              /* gpm information      */
 
 /* display resulting data */
 int display_data(Gpm_Event *event, void *data)
diff --git a/src/prog/get-versions.c b/src/prog/get-versions.c
index 00f1753..8abd38d 100644
--- a/src/prog/get-versions.c
+++ b/src/prog/get-versions.c
@@ -22,7 +22,7 @@
  ********/
 
 #include <stdio.h>            /* printf()             */
-#include <gpm.h>              /* gpm information      */
+#include "headers/gpm.h"              /* gpm information      */
 
 int main()
 {
diff --git a/src/prog/gpm-root.y b/src/prog/gpm-root.y
index 069d801..6b3a5a6 100644
--- a/src/prog/gpm-root.y
+++ b/src/prog/gpm-root.y
@@ -443,6 +443,7 @@ void f__fix(struct passwd *pass)
 }
 
 /*---------------------------------------------------------------------*/
+#if 0
 static int f_debug_one(FILE *f, Draw *draw)
 {
    DrawItem *ip;
@@ -465,6 +466,7 @@ static int f_debug_one(FILE *f, Draw *draw)
 #undef LINE
    return 0;
 }
+#endif
 
 int f_debug(int mode, DrawItem *self, int uid)
 {
@@ -960,10 +962,8 @@ static inline void scr_dump(int fd, FILE *f, unsigned char *buffer, int vc)
 /*------------*/
 static inline void scr_restore(int fd, FILE *f, unsigned char *buffer, int vc)
 {
-   int x,y, dumpfd;
+   int dumpfd;
    char dumpname[20];
-
-   x=buffer[2]; y=buffer[3];
    
    /* WILL NOT WORK WITH DEVFS! FIXME! */
    sprintf(dumpname,"/dev/vcsa%i",vc);
@@ -1196,11 +1196,7 @@ int main(int argc, char **argv)
                                                         LOG_DAEMON : LOG_USER);
    /* reap your zombies */
    childaction.sa_handler=reap_children;
-#if defined(__GLIBC__)
-   __sigemptyset(&childaction.sa_mask);
-#else /* __GLIBC__ */
-   childaction.sa_mask=0;
-#endif /* __GLIBC__ */
+   sigemptyset(&childaction.sa_mask);
    childaction.sa_flags=SA_INTERRUPT; /* need to break the select() call */
    sigaction(SIGCHLD,&childaction,NULL);
 
-- 
2.52.0

